src/mlpack/core/cv/k_fold_cv_impl.hpp

```diff
@@ -269,6 +284,58 @@ double KFoldCV<MLAlgorithm,
   return arma::mean(evaluations);
 }
 
+template<typename MLAlgorithm,
+         typename Metric,
+         typename MatType,
+         typename PredictionsType,
+         typename WeightsType>
+template<bool Enabled, typename>
+void KFoldCV<MLAlgorithm,
+             Metric,
+             MatType,
+             PredictionsType,
+             WeightsType>::ShuffleData()
+{
+  MatType data = xs.cols(0, (k - 1) * binSize + lastBinSize - 1);
+  PredictionsType labels = ys.subvec(0, (k - 1) * binSize + lastBinSize - 1);
```

Comment:
Probably it's better to use the name "predictions" instead of "labels" since k-fold cross-validation can be applied to such methods as `LinearRegression`. Also probably it's better to use the method `cols` rather than `subvec` if one day we want to use it with something like `FFN` (which in the current implementation uses `arma::mat` for `responses`).

Reply:
Actually I'll just change them to `xsNew` and `ysNew`, there is no confusion there.  And you are right, `cols()` would be better.